static/js/containers/ProfileImage.js

```diff
@@ -68,27 +72,29 @@ class ProfileImage extends React.Component<*> {
   }
 
   render() {
-    const { profile, useSmall, editable, onClick } = this.props
+    const { profile, imageSize, editable, onClick } = this.props
 
-    const imageUrl = makeProfileImageUrl(profile, useSmall)
-    const imageSizeClass = useSmall ? "small" : "medium"
+    const imageUrl = makeProfileImageUrl(
+      profile,
+      imageSize === "micro" || imageSize === "small"
```

Comment:
I was on the fence when I was writing it, on the one hand it would prevent typos, but Flow *should* do that automatically. One thing that is annoying about these union types is that it doesn't seem to allow you to declare constants (as JS variables) and then declare a type using those variables. I wanted to do something like this:

```js
const PROFILE_IMAGE_MICRO = "micro"
const PROFILE_IMAGE_SMALL = "small"
const PROFILE_IMAGE_MEDIUM = "medium"

type ImageSize =
  | PROFILE_IMAGE_MICRO
  | PROFILE_IMAGE_SMALL
  | PROFILE_IMAGE_MEDIUM
```

But when I tried it gave me an error about needing to use the `typeof` operator - but when I do that the type for `ImageSize` ends up being `string | string | string`! With the current setup for the `ImageSize` type it throws a type error if I pass, for instance, "Micro" as the `imageSize` prop, but if the type is `string | string | string` it will obviously accept any string.

I'm not sure if this is a limitation for Flow or I just don't know the right syntax. Anyway, so we could declare constants for the three strings, which would make it easier to change them in one place later if we wanted, but it doesn't seem like they can be used in the flow type :/

Reply:
this provides a little context for that: https://github.com/facebook/flow/issues/2639

Reply:
I'll go ahead and declare constants to use when setting the size prop though